File: Binary二进制/leetcode_693_hasAlternatingBits.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool hasAlternatingBits(int n) {
        while ((n&1) != ((n>>1)&1)){
            n >>= 1;
        }
        return n==0;
    }
};

int main(){
    Solution s;
    int n = 5;
    cout << s.hasAlternatingBits(n) << endl;
}

File: 02.Binary二进制/leetcode_693_hasAlternatingBits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        while n&1 != (n>>1)&1:
            n >>= 1
        return n==0

if __name__ == "__main__":
    s = Solution()
    n = 5
    print(s.hasAlternatingBits(n))
